### ISSUE TITLE
Magazines add volume to pistols

### ISSUE THREAD

**Issue by TheGamingFox1911** [CleverRaven/Cataclysm-DDA#41256]
# Describe the bug

Magazines add volume to pistols, like the Berreta Px4 Storm, when the magazine is inside the grip of the gun.

# Steps To Reproduce

1. Get a pistol that takes magazines and the right magazine for it.
2. look at the volume of the weapon loaded and unloaded

# Expected behavior

Magazines should not add to the volume of the weapon. 

# Screenshots

https://cdn.discordapp.com/attachments/598523535169945607/721055229931683970/unknown.png
https://cdn.discordapp.com/attachments/598523535169945607/721055233228144810/unknown.png

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.E-3101-g483a50f [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Aftershock [aftershock],
    C.R.I.T Expansion Mod [crt_expansion],
    Magiclysm [magiclysm],
    DinoMod [DinoMod],
    Mythical Martial Arts [MMA],
    Bionic Slots [cbm_slots],
    Disable NPC Needs [no_npc_food],
    Fuji's Military Profession Pack [fuji_mpp]
]

# Additional context

A loaded magazine doesn't seem to add to the weight of the gun, just the volume.

**Comment by wapcaplet** [comment]
Confirmed; separately, the Px4 is 0.51 L, and the magazine 0.34 L:

![image](https://user-images.githubusercontent.com/238395/84555141-93c07b00-acd8-11ea-9dc8-cfd2f30fad5e.png)

When loaded, they total to 0.84 L:

![image](https://user-images.githubusercontent.com/238395/84555154-a3d85a80-acd8-11ea-9857-ff728dc2474b.png)

- OS: Linux
    - OS Version: Distributor ID: Ubuntu; Description: Pop!_OS 19.10; Release: 19.10; Codename: eoan; 
- Game Version: 0.E-3128-gb880329a5e [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food]
]

**Comment by Tevervin** [comment]
That is the issue with every weapon, no matter how it reloads - with magazines or only with bullets. Bullets volume adds to magazine (or straight to the gun, in case of most shotguns and similar guns), than combined volume of bullets and magazine adds to the gun.

Bullets should not change volume of guns (if it not RPG-7, speargun or something), but magazines are difficult case. Pistol magazines fully fits inside gun's handle, but there is extended variants that should partially increase gun's volume. Magazines of most rifles are partially covered by rifle when attached to it, so not all magazine volume should be added to rifle's volume, but to implement how much volume of every magazine should not adds to every gun is pretty big task, in my opinion.

As for bullets and mags, mb it is issue of nested containers and mags should be considered "not rigid" to not increase their volume?

**Comment by ymber** [comment]
The problem is the pockets not being rigid. I quickly automated them for infrastructure work but they need auditing for content standards. The numbers for mag wells are way out right now.

**Comment by KorGgenT** [comment]
pockets not being rigid but with an added `magazine_well` volume is the intended way to deal with it. should be a json fix, but it'll be a lot of json editing.

**Comment by wapcaplet** [comment]
To clarify, guns with a `MAGAZINE_WELL` pocket ought to define a `"magazine_well"` volume, to define the largest magazine that will fit without sticking out.

For example, to give the Beretta Px4 Storm a magazine well large enough to contain the standard 250 ml magazine, you'd set a "magazine_well" volume in the gun's "MAGAZINE_WELL" pocket:

```json
"id": "px4",
"type": "GUN",
"pocket_data": [
  {
    "pocket_type": "MAGAZINE_WELL",
    "//": "magazines over 250 ml will stick out. Only works if rigid is false (the default)"
    "magazine_well": "250 ml"
  }
]
```

This means that any magazine up to 250 ml will not add to the gun's volume, but magazines larger than that will stick out and increase the volume.

The magazine itself could be handled similarly, but I assume it's less likely for magazine volume to be increased by ammo sticking out the end, so those could be modeled by making them "rigid" instead:

```json
"id": "px4mag",
"type": "MAGAZINE",
"volume": "250 ml",
"pocket_data": [
  {
    "pocket_type": "MAGAZINE",
    "//": "this magazine is rigid, and always 250 ml whether empty or full",
    "rigid": true
  }
]
```

**Comment by meladath** [comment]
@wapcaplet You are correct. I attempted to fix this issue in my ex PR that made all magazine wells rigid, which I submitted the same time as #41038. That was "fine" on my end for quite a while and allowed guns to be holstered. However no gun increased in volume with extended magazines.

Once #41123 was merged however the "solution" of making all magazine wells rigid causes the UI to display 20L per every gun in your inventory (since every MAGAZINE_WELL atm is 20L). So that is no longer a viable solution.

The only "fix" is lots of JSON magazine_well entries, combined with more json for magazine sizes.
